docs/design/proposals/stateless-csi-plugin.md

```diff
@@ -0,0 +1,507 @@
+  - Some of the RPCs as of v1 of the CSI specification that would hence
+  not be supported are,
+    - Listing volumes and snapshots
+    - Getting capacity of the backing Ceph cluster(s)
+
+### Configuration specifics
+
+#### `config.json` config map contents
+
+```json
+{
+  "<clusterID>": {
+    "monitors": [
+      "IP/DNS:port",
+      ...
+    ]
+  },
+  ...
+}
+```
+
+#### Annotated YAML for RBD StorageClass
+
+```yaml
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: fast
+provisioner: csi-rbd
+parameters:
+  # [Optional] defaults to "ext4" allowed values depend on filesystems
+  # supported
+  fsType: ext4
+  # [Optional] defaults to "2"
+  imageFormat: "2"
+  # [Optional], can be specified when imageFormat is used, defaults to "",
+  #     possible values "layering"
+  imageFeatures: "layering"
+  # [Mandatory] carries the fsid of the Ceph cluster
+  clusterID: <ID>
+  # [Mandatory] carries the pool name to use in the Ceph cluster defined
+  #     by clusterID
+  pool: <pool-name>
+  # [Mandatory] All of the following are optional, and as described in,
+  #     https://kubernetes-csi.github.io/docs/secrets-and-credentials.html
+  csi.storage.k8s.io/provisioner-secret-name: csi-rbd-secret
+  csi.storage.k8s.io/provisioner-secret-namespace: default
+  csi.storage.k8s.io/node-publish-secret-name: csi-rbd-secret
+  csi.storage.k8s.io/node-publish-secret-namespace: default
+```
+
+#### Annotated YAML for CephFS StorageClass
+
+**NOTE:** For terseness only deltas are detailed below
+
+```yaml
+  # [Mandatory]
+  clusterID: <ID>
+  # [Mandatory] carries the filesystem name to use in the Ceph cluster defined
+  #     by clusterID
+  fsname: <fs-name>
+
+  provisionVolume: "true"
+  # Ceph pool into which the volume shall be created
+  # Required for provisionVolume: "true"
+  pool: cephfs_data
+
+  # Root path of an existing CephFS volume
+  # Required for provisionVolume: "false"
+  # rootPath: /absolute/path
+```
+
+### VolumeID and Ceph image/sub-directory naming
+
+This section details how we will deal with image/FS names on Ceph cluster, when
+provisioning volumes based on *volume_name* passed in by the CO, in the
+CreateVolume request to the CS.
+
+As per the CSI spec, creation requests come with a *volume_name* that is
+controlled by the CO and is unique across different volume requests, and the
+same for a given volume to maintain idempotent create requests.
+
+The CSI plugin returns a *volume_id* that is passed to other plugin RPCs to
+refer to the volume, and is generated by the plugin. Snippet from the spec
+about the *volume_id*,
+
+```
+  // This field MUST contain enough information to uniquely identify
+  // this specific volume vs all other volumes supported by this plugin.
+  // This field SHALL be used by the CO in subsequent calls to refer to
+  // this volume.
+```
+
+#### Name terminology
+
+- **ImageName**: Name of the image as created on Ceph (rbd image name, or
+  CephFS sub-directory)
```

Comment:
sub-directory name?

Reply:
Yes, using the same keyword ImageName to denote a RBD image name or the name of a CephFS sub-directory. Is there something missing?